docs/custom-metric-filter.md

```diff
@@ -0,0 +1,151 @@
+Config Sync will try to apply the Open Telemetry Collector ConfigMap that contains
+[googlecloud exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/googlecloudexporter)
+to write metrics into Cloud Monitoring. The default configuration contains a
+bare minimum list of metrics, but some customers might want to configure
+different backends or fine tune the metric list for better monitoring solutions.
+
+This guide includes instructions for how to adjust filters, modify exporters, 
+or opt out of exporting metrics all together.
+
+Note: This workaround applies to Config Sync v1.8+, when custom monitoring
+was integrated.
+
+## Default metrics pipeline config (v1.12+)
+
+Config Sync will check for the default credential in the environment when
+starting up. For a GKE cluster which always have a default service account,
+Config Sync will deploy the following Open Telemetry Collector configuration to
+enable the Google Cloud monitoring pipelines.
+
+A pipeline consist of:
+
+-   receivers: receives metrics from all reconcilers inside Config Sync
+-   processors: filters or transforms metrics before sending out
+-   exporters: various backend endpoints and their configurations
+
+Config Sync uses
+[filter processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/filterprocessor)
+to control the exclusive list of metrics that goes into each monitoring backend.
+
+Config Sync uses two major exporters: [googlecloud exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/googlecloudexporter)
+[prometheus exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/prometheusexporter)
+
+- `googlecloud` stands for the configuration that exports to the Cloud
+Monitoring backend.
+- `googlecloud/kubernetes` stands for the configuration that exports to the
+Cloud Monarch backend
+- `prometheus` stands for the configuration that exports metrics in Prometheus
+format and later can be scraped by a prometheus server following [this guide](http://cloud/anthos-config-management/docs/how-to/monitoring-config-sync#prometheus)
+
+Here is [the current configuration](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/metrics/otel.go#L38) of Open Telemetry Collector
+in Config Sync.
+
+## Patch the otel collector deployment with ConfigMap
+
+1. **Get the current ConfigMap as template**
+
+    ```
+    kubectl get cm otel-collector-googlecloud -n config-management-monitoring -o yaml > otel-collector-config.yaml
+    ```
+    
+    Or in Config Sync with version prior to `v1.11.1`, use
+    
+    ```
+    kubectl get cm otel-collector-stackdriver -n config-management-monitoring -o yaml > otel-collector-config.yaml
+    ```
+1. Change the `name` under `.data.otel-collector-config.yaml` to `otel-collector-custom`
```

Comment:
what's the `.data.` for?

Reply:
The data field in the fetched ConfigMap.

Reply:
I'm confused. Isn't it just the `.metadata.name` of the ConfigMap we want to change? Or do you need to modify the data string value?

Reply:
Oh! I was looking at wrong spot, fixing in a minute, sorry for the confusion!

Reply:
Done with latest commit